packages/@aws-cdk/aws-iam/lib/role.ts

```diff
@@ -219,19 +224,26 @@ export interface ImportedRoleProps {
    * The role's ARN
    */
   roleArn: string;
+
+  /**
+   * The role's Name
+   */
+  roleName: string;
```

Comment:
I am not sure that making both ARN and name required is the best developer experience. In many cases one can be deduced from the other. See example in [ECR](https://github.com/awslabs/aws-cdk/blob/9a7709e5fa97c4c5aa0484fbbd041291efaa3bcf/packages/%40aws-cdk/aws-ecr/lib/repository-ref.ts#L68), where both are actually optional and one is deduced from the other in certain cases.

Not 100% sure that's the ideal situation, but I wouldn't want to require both if they can be deduced.

Reply:
Sounds good.  I will take a look.